libs/x-content/src/main/java/org/opensearch/common/xcontent/XContent.java

```diff
@@ -46,7 +46,7 @@ public interface XContent {
     /**
      * The type this content handles and produces.
      */
-    XContentType type();
+    MediaType type();
```

Comment:
I believe the code was right before,  because this is `XContent`, it is expected to have `XContentType`. You may consider to introduce `MediaType mediaType();`.

Reply:
XContent base class is going to be refactored to `opensearch-core` library and the concrete implementation is going to `opensearch-xcontent` library. That means `XContentType` is going to the xcontent library but XContent will be going to core... So this has to be separated. It's not perfect but it's necessary for package separation. We'll need to separate XContent and Media further in a follow up.

Reply:
What I can do is open another PR that further introduces `MediaContent`, `MediaParser`, `MediaGenerator`, etc, and then have the `XContent*` interfaces derive from those base classes, I was hoping to introduce that abstraction in a followup PR, however, so that we can first accomplish the minimal separation needed to trim down the server classes. Make sense?

Reply:
Another idea would be to refactor all of the existing `XContent`, `XContentBuilder`, `XContentFactory`, etc. interfaces as `MediaContent`, `MediaContentBuilder`, etc, (do away w/ the "man in the middle" classes) then refactor those to `opensearch-core` library under `org.opensearch.core.mediacontent` and keep the  `CBOR`, `JSON`, `YAML`, `SMILE` concrete implementations in the `opensearch-xcontent` library. Either way, in order to move forward on refactoring server components (e.g., aggregations) we need to sever all of this tight xContent coupling from the base classes. I'd like to incrementally and organically make progress there than over design up front. Which ever way we go, it's not a one way door.

Reply:
Ok, I see, thanks @nknize , let's keep it as simple as possible: what about replacing `XContentType type();` with ` MediaType mediaType()`? That would be clear that this is indeed media type.